app/interactors/access_limit/update_interactor.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+class AccessLimit::UpdateInteractor < ApplicationInteractor
+  delegate :params,
+           :user,
+           :edition,
+           to: :context
+
+  def call
+    Edition.transaction do
+      find_and_lock_edition
+      update_edition
+      create_timeline_entry
+    end
+  end
+
+private
+
+  def find_and_lock_edition
+    context.edition = Edition.find_current(document: params[:document])
```

Comment:
we seem to be missing the lock on this

Reply:
Oops - this alerted me to an unused method: Edition#find_and_lock_current. I've added a commit to remove it.

Reply:
Cool - sounds good 👍